cmd/file.d_e2e_new_test.go

```diff
@@ -0,0 +1,253 @@
+//go:build e2e_new
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	http2 "net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ozontech/file.d/cfg"
+	"github.com/ozontech/file.d/fd"
+	"github.com/ozontech/file.d/test"
+	uuid "github.com/satori/go.uuid"
+	"github.com/segmentio/kafka-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+/*
+General interface for e2e tests
+AddConfigSettings - function that preparing а config for your test
+Send - function that send message in pipeline and waits for the end of processing
+Validate - function that validate result of the work
+*/
+
+type e2eTest interface {
+	AddConfigSettings(t *testing.T, conf *cfg.Config, pipeName string)
+	Send(t *testing.T)
+	Validate(t *testing.T)
+}
+
+func startForTest(t *testing.T, e e2eTest, configPath, pipeName string) *fd.FileD {
+	conf := cfg.NewConfigFromFile(configPath)
+	e.AddConfigSettings(t, conf, pipeName)
+	filed := fd.New(conf, "off")
+	filed.Start()
+	return filed
+}
+
+func TestE2EStabilityWorkCase(t *testing.T) {
+	testsList := []struct {
+		e2eTest
+		cfgPath  string
+		pipeName string
+	}{
+		{
+			e2eTest: &fileFile{
+				outputNamePattern: "/file-d.log",
+				filePattern:       "pod_ns_container-*",
+				count:             10,
+				lines:             500,
+				retTime:           "1s",
+			},
+			cfgPath:  "./../testdata/config/e2e_cfg/file_file.yaml",
+			pipeName: "test_file_file",
```

Comment:
why you need to provide pipeline name here? you already has pipeline in the configuration file